### PR TITLE
Benchmark GHA

### LINKED ISSUE #13: Add GA benchmarks

### PR DESCRIPTION
**Pull request by mwiewior**
Resolves #13

### CODE REVIEW

File: sequila/sequila-core/benches/databio_benchmark.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        // Table { name: String::from("chainOrnAna1"), non_flat: 6 },
+        // Table { name: String::from("chainVicPac2"), non_flat: 8 },
+        // Table { name: String::from("chainXenTro3Link"), non_flat: 7 },
+        // Table { name: String::from("chainMonDom5Link"), non_flat: 7 },
+        // Table { name: String::from("ex-anno"), non_flat: 2 },
+        // Table { name: String::from("ex-rna"), non_flat: 7 }
+    ];
+
+    const QUERY: &str = r#"
+            select count(*) from s1 a, s2 b  where (a.contig=b.contig and a.pos_end>=b.pos_start and a.pos_start<=b.pos_end)
```
[the remark this answers is not in the document]
Comment (pkhamutou) [reply sequila/sequila-core/benches/databio_benchmark.rs]: how do we add datasets there?

**Comment by pkhamutou** [comment]
btw, where do we publish it?

File: sequila/sequila-core/benches/databio_benchmark.rs [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+    s2: Table,
+}
+
+fn get_test_name(
+    algorithm: Option<Algorithm>,
+    t1: Table,
+    t2: Table,
+    prefix: Option<&String>,
+) -> String {
+    let test_name = String::from(match prefix {
```
Comment (pkhamutou) [inline sequila/sequila-core/benches/databio_benchmark.rs:39]: i'd simplify it:
```rust
    let prefix = prefix
        .map(|p| p.to_string() + "-")
        .unwrap_or("".to_string());

    format!(
        "{}{}-{}-{}-{}-{}-{}-{}",
        prefix,
        algorithm.unwrap(),
        t1.table_size,
        t2.table_size,
        t1.non_flat,
        t2.non_flat,
        t1.id,
        t2.id
    )
```

File: sequila/sequila-core/benches/databio_benchmark.rs [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+
+    SessionContext::new_with_sequila(config)
+}
+
+pub fn databio_benchmark(c: &mut Criterion) {
+    let root_path = env::var("BENCH_DATA_ROOT").unwrap_or("/data/bench_data/databio/".to_string());
+    let tables = [
+        Table {
+            id: 0,
+            name: String::from("chainRn4"),
```
Comment (pkhamutou) [inline sequila/sequila-core/benches/databio_benchmark.rs:156]: i think `"chainRn4".to_string()` reads better :)

File: sequila/sequila-core/benches/databio_benchmark.rs [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+            s1: tables[0].clone(),
+            s2: tables[8].clone(),
+        },
+        TestCase {
+            s1: tables[7].clone(),
+            s2: tables[8].clone(),
+        },
+    ];
+    let algorithms = [
+        Some(Algorithm::Coitrees),
```
Comment (pkhamutou) [inline sequila/sequila-core/benches/databio_benchmark.rs:247]: why is it an Option? I think it's never `None`

File: sequila/sequila-core/benches/databio_benchmark.rs [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+        .build()
+        .unwrap();
+
+    let mut group = c.benchmark_group("Algorithms");
+    group.sample_size(10);
+    group.measurement_time(std::time::Duration::from_secs(7200));
+    group.significance_level(0.1);
+
+    for test in test_cases {
+        let s1_path = root_path.clone() + &test.s1.name + &String::from("/*.parquet");
```
Comment (pkhamutou) [inline sequila/sequila-core/benches/databio_benchmark.rs:262]: i think it's better to prefer `format!()` for string concatenations:
```rust
        let s1_path = format!("{}{}{}", root_path, test.s1.name, "/*.parquet");
```

File: sequila/sequila-core/benches/databio_benchmark.rs [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+        let s2_path = root_path.clone() + &test.s2.name + &String::from("/*.parquet");
+        let test_baseline = get_test_name(
+            Some(Algorithm::Coitrees),
+            test.s1.clone(),
+            test.s2.clone(),
+            Some(&String::from("baseline")),
```
Comment (pkhamutou) [inline sequila/sequila-core/benches/databio_benchmark.rs:268]: why is it a reference?

**Review by pkhamutou — APPROVED** [review]
Good job! 

You can ignore my comments, it's mostly about Rust, we don't really care about it here.

File: sequila/sequila-core/benches/databio_benchmark.rs (hunk 4 above)
Comment (mwiewior) [reply sequila/sequila-core/benches/databio_benchmark.rs:247]: Fixed.

File: sequila/sequila-core/benches/databio_benchmark.rs (hunk 5 above)
Comment (mwiewior) [reply sequila/sequila-core/benches/databio_benchmark.rs:262]: Fixed

File: sequila/sequila-core/benches/databio_benchmark.rs (hunk 6 above)
Comment (mwiewior) [reply sequila/sequila-core/benches/databio_benchmark.rs:268]: Fixed.

**Comment by mwiewior** [comment]
> Good job!
> 
> You can ignore my comments, it's mostly about Rust, we don't really care about it here.

I applied almost all of them !
Thanks.

**Pull request merged**